alicloud/resource_alicloud_cas.go

```diff
@@ -0,0 +1,116 @@
+	return &schema.Resource{
+		Create:   resourceAlicloudCasCreate,
+		Read:     resourceAlicloudCasRead,
+		Update:   resourceAlicloudCasUpdate,
+		Delete:   resourceAlicloudCasDelete,
+		Importer: &schema.ResourceImporter{State: resourceAlicloudCasImporter},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateCasName,
+			},
+			"cert": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCasCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	args := cas.CreateCreateUserCertificateRequest()
+	if v, ok := d.GetOk("name"); ok {
+		args.Name = v.(string)
+	} else {
+		return fmt.Errorf("Need param Name")
+	}
+
+	if v, ok := d.GetOk("cert"); ok {
+		b, err := ioutil.ReadFile(v.(string))
+		if err != nil {
+			return fmt.Errorf("OpenCertificate key file got an error: %#v", err)
+		}
+		args.Cert = string(b)
+	} else {
+		return fmt.Errorf("Need param Cert")
+	}
+
+	if v, ok := d.GetOk("key"); ok {
+		b, err := ioutil.ReadFile(v.(string))
+		if err != nil {
+			return fmt.Errorf("OpenCertificate key file got an error: %#v", err)
+		}
+		args.Key = string(b)
+	} else {
+		return fmt.Errorf("Need param Key")
+	}
+
+	raw, err := client.WithCasClient(func(casClient *cas.Client) (interface{}, error) {
+		return casClient.CreateUserCertificate(args)
+	})
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("CreateUserCertificate got an error: %#v", err)
+	}
+	response, _ := raw.(*cas.CreateUserCertificateResponse)
+	d.SetId(string(response.CertId))
+	return resourceAlicloudCasRead(d, meta)
+}
+
+func resourceAlicloudCasUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceAlicloudCasRead(d, meta)
+}
+
+func resourceAlicloudCasRead(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceAlicloudCasDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	args := cas.CreateDeleteUserCertificateRequest()
+	args.CertId = requests.Integer(d.Id())
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := client.WithCasClient(func(casClient *cas.Client) (interface{}, error) {
+			return casClient.DeleteUserCertificate(args)
+		})
+		if err != nil {
+			if IsExceptedError(err, CertNotExist) {
+				return resource.RetryableError(fmt.Errorf("Operation failt - trying again after change complete."))
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error deleting cert %s: %#v", d.Id(), err))
+		}
+		return nil
+	})
+}
+
+func resourceAlicloudCasImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	return nil, nil
```

Comment:
Please add testcases for the datasource and resource.

Reply:
done